web/src/hashchange.js

```diff
@@ -367,6 +369,15 @@ function do_hashchange_overlay(old_hash) {
     if (base === "scheduled") {
         scheduled_messages_overlay_ui.launch();
     }
+    if (base === "user") {
+        const user_id = Number.parseInt(hash_parser.get_current_hash_section(), 10);
+        if (!people.is_known_user_id(user_id)) {
+            user_profile.show_user_profile_access_error_modal();
+        } else {
+            const user = people.get_by_user_id(user_id);
+            user_profile.show_user_profile(user);
+        }
+    }
```

Comment:
What happens for logged-out users (`page_params.is_spectator`)? Some similar code paths call the `login_to_access` function -- grep for it to find examples.

Reply:
The implementation to handle logged-out users is already being managed by the `hashchanged` function. Since we navigate to the user profile by passing through the hashchanged function, it triggers the `login_to_access` profile modal when a logged-out user attempts to navigate to a specific user profile URL.